src/com/rcv/ResultsWriter.java

```diff
@@ -99,13 +102,27 @@ ResultsWriter setElectionConfig(ElectionConfig config) {
     return this;
   }
 
+  // function: setTimestamp
+  // purpose: setter for timestamp string used for creating output file names
+  // param: timestamp string to use for creating output file names
+  ResultsWriter setTimestamp(String timestamp) {
+    this.timestamp = timestamp;
+    return this;
```

Comment:
Confused as to why we're returning a ResultsWriter here... shouldn't this just be a void method? I guess this is for chaining purposes... is this standard practice?

Reply:
We've been doing this for a bunch of ResultsWriter setters. Not sure how common or uncommon it is for Java, but I find it makes the constructor callsite more readable if you have more than a few args you need to supply. It does bloat the class code a bit, though.

Reply:
Confirmed that this seems to be standard practice, so 👍

Reply:
While common practice I find this style weird (setters returning this???).  But whatever.  My hot take is, use multiple parameters (real coders can handle that level or responsibility ;) and just put comments in the call site when it gets confusing.